samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -160,8 +169,8 @@ public String getChangelogSystem() {
    * @param storeName name of the store
    * @return the class name of SideInputsProcessorFactory if present, null otherwise
```

Comment:
Would be better to update the java doc to indicate that this method returns Optional.empty if the side-input factory is not defined.

Reply:
Good catch